README.md

```diff
@@ -1,130 +1,181 @@
-import * as wn from 'webnative'
+// ESM
+import * as wn from "webnative"
 
 // Browser/UMD build
 const wn = globalThis.webnative
 ```
 
-# Authentication
+## Creating a Program
+
+A Webnative program is an assembly that makes up a part of, or a whole distributed web application. It consists out of several components that can be customised. _We'll stick with the default components for now, which means we'll be using the Web Crypto API._
 
 ```ts
-const state = await wn.initialise({
-  permissions: {
-    // Will ask the user permission to store
-    // your apps data in `private/Apps/Nullsoft/Winamp`
-    app: {
-      name: "Winamp",
-      creator: "Nullsoft"
-    },
-
-    // Ask the user permission to additional filesystem paths
-    fs: {
-      private: [ wn.path.directory("Audio", "Music") ],
-      public: [ wn.path.directory("Audio", "Mixtapes") ]
-    }
+const program = await wn.program({
+  // Can also be a string, used as an identifier for caches.
+  // If you're developing multiple apps on the same localhost port,
+  // make sure these differ.
+  id: { creator: "Nullsoft", name: "Winamp" }
+
+}).catch(error => {
+  switch (error) {
+    case webnative.ProgramError.InsecureContext:
+      // Webnative requires HTTPS
+      break;
+    case webnative.ProgramError.UnsupportedBrowser:
+      break;
   }
 
-}).catch(err => {
-  switch (err) {
-    case wn.InitialisationError.InsecureContext:
-      // We need a secure context to do cryptography
-      // Usually this means we need HTTPS or localhost
+})
+```
 
-    case wn.InitialisationError.UnsupportedBrowser:
-      // Browser not supported.
-      // Example: Firefox private mode can't use indexedDB.
-  }
+That gives us a `Program` object, with this we can create a new user session or reuse the existing session. There's two distinctive ways to create a user session, either by using an authentication strategy or by using something we call "confidences". More on the latter in a bit, let's start with the default authentication strategy.
 
-})
+```ts
+let session
+
+// Do we have an existing session?
+if (program.session) {
+  session = program.session
+
+// If not, let's authenticate.
+// (a) new user, register a new Fission account
+} else if (registerNewUser) {
+  const { success } = await program.auth.register({ username: "llama" })
+  session = success ? program.auth.session() : null
+
+// (b) existing user, link a new device
+} else {
+  // On device with existing session:
+  const producer = program.auth.accountProducer(program.session.username)
+
+  producer.on("challenge", challenge => {
+    // Either show `challenge.pin` or have the user input a PIN and see if they're equal.
+    if (userInput === challenge.pin) challenge.confirmPin()
+    else challenge.rejectPin()
+  })
+
+  producer.on("link", ({ approved }) => {
+    if (approved) console.log("Link device successfully")
+  })
+
+  // On device without session:
+  //     Somehow you'll need to get a hold of the username.
```

Comment:
small typo. `ahold` is one word in this context :) 
```suggestion
  //     Somehow you'll need to get ahold of the username.
```

Reply:
Good to know, thanks 😄